src/test/java/api/loans/anonymization/AnonymizeLoansAfterXIntervalTests.java

```diff
@@ -361,6 +369,67 @@ public void testClosedLoansAnonymizedAfterIntervalPassed() {
         .getJson(), not(isAnonymized()));
   }
 
+  @Test
+  public void doNotAnonymizeLoansTwice() {
+    createAnonymizeAfterIntervalConfiguration(1, "minute");
+
+    final InventoryItemResource nod = itemsFixture.basedUponNod();
+    final InventoryItemResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+
+    final IndividualResource firstLoan = loansFixture.checkOutByBarcode(smallAngryPlanet, user);
+    final IndividualResource secondLoan = loansFixture.checkOutByBarcode(nod, usersFixture.rebecca());
+
+    loansFixture.checkInByBarcode(smallAngryPlanet);
+    loansFixture.checkInByBarcode(nod);
+
+    setNextAnonymizationDateTime(ONE_MINUTE_AND_ONE);
+
+    LoanAnonymizationAPIResponse firstAnonymization = anonymizeLoansInTenant();
+    assertThat(firstAnonymization.getAnonymizedLoans().size(), is(2));
+    assertThat(loansStorageClient.getById(firstLoan.getId()).getJson(), isAnonymized());
+    assertThat(loansStorageClient.getById(secondLoan.getId()).getJson(), isAnonymized());
+
+    setNextAnonymizationDateTime(ONE_MINUTE_AND_ONE);
+
+    LoanAnonymizationAPIResponse secondAnonymization = anonymizeLoansInTenant();
+    assertThat(secondAnonymization.getAnonymizedLoans().size(), is(0));
+  }
+
+  @Test
+  public void ignoresAlreadyAnonymizedLoans() {
```

Comment:
How does this demonstrate that the already anonymized loans are ignored?

If the system fetches loans for anonymization in batches of larger than 2, won't the second two loans be picked up irrespective of whether the first 2 are ignored or not?

Reply:
The test demonstrates following steps:

1. Check out 4 items;
1. Close loans for 1 and 2 items (out of 4), via check in;
1. Run the anonymization process;
1. Verify that loans for the 2 items have been anonymized;
1. Close loans for 3 and 4 items, via check in;
1. Run the anonymization process again;
1. Verify that we only processed loans for 3 and 4 items and loans 1 and 2 have not been fetched again (the anonymization endpoint returns list of ids of processed loans).

Reply:
Ok, thank you for explaining it to me. 

I don't have time at the moment to investigate the changes in detail.